### PR TITLE
add keepa gateway

### PR DESCRIPTION
**Pull request by farshadth**

### CODE REVIEW

File: poolport-sample.php [hunk 1]
```diff
@@ -231,12 +231,21 @@
...
+
     // Dara gateway
     //--------------------------------
     'dara'          => array(
-        'merchant-id'  => 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
-        'terminal-key' => 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
-        'terminal-id'  => 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        'merchant-id'  => 120000000185396,
+        'terminal-key' => 120000000185396,
+        'terminal-id'  => 3017273953,
```
Comment (m-jch) [inline poolport-sample.php:240]: Do not add sensitive data in git history

File: src/Keepa/Keepa.php [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    {
+        $this->newTransaction();
+
+        try {
+            $client = new Client();
+
+            $response = $client->request("POST", "{$this->gateUrl}/request_payment_token", [
+                "json"    => [
+//                    'amount'       => $this->amount,
+                    'amount'       => 1000,
```
Comment (m-jch) [inline src/Keepa/Keepa.php:96]: Amount is not correct

File: src/Keepa/Keepa.php [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+            if ($response->Status != 200) {
+                $this->transactionFailed();
+                $this->newLog($response->Status, $response->Message);
+                throw new KeepaException($response->Message, $response->Status);
+            }
+
+            $this->token = $response->Content->payment_token;
+            $this->paymentUrl = $response->Content->payment_url;
+            $this->refId = $this->transactionId();
```
Comment (m-jch) [inline src/Keepa/Keepa.php:115]: We can use $this->token as refId, don't use $this->transactionId();

File: src/Keepa/Keepa.php [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            $this->transactionSetRefId();
+
+            $this->setMeta([
+                'token'     => $this->token,
```
Comment (m-jch) [inline src/Keepa/Keepa.php:119]: If we use $this->token as refId we don't need to  set it here as meta anymore

File: src/Keepa/Keepa.php [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+     */
+    protected function verifyPayment()
+    {
+        try {
+            $client = new Client();
+
+            $response = $client->request("POST", "{$this->gateUrl}/verify_transaction", [
+                "json"    => [
+                    'payment_token'  => $this->getMeta('token'),
+                    'reciept_number' => $this->getMeta('reciept_number'),
```
Comment (m-jch) [inline src/Keepa/Keepa.php:144]: I can not find where we set reciept_number in meta

**Review by m-jch — CHANGES_REQUESTED** [review]

File: src/Keepa/Keepa.php (hunk 4 above)
Comment (farshadth) [reply src/Keepa/Keepa.php:119]: no token is apart from refId, refId should be the id of the response in order to track specific transaction in further

File: src/Keepa/Keepa.php (hunk 5 above)
Comment (farshadth) [reply src/Keepa/Keepa.php:144]: reciept_number is posted from the gateway in callback, it is set in project

**Pull request closed without merging**